stan/math/prim/mat/prob/inv_wishart_rng.hpp

```diff
@@ -1,39 +1,30 @@
 #ifndef STAN_MATH_PRIM_MAT_PROB_INV_WISHART_RNG_HPP
 #define STAN_MATH_PRIM_MAT_PROB_INV_WISHART_RNG_HPP
 
-#include <stan/math/prim/mat/err/check_ldlt_factor.hpp>
-#include <stan/math/prim/scal/err/check_greater.hpp>
-#include <stan/math/prim/scal/err/check_size_match.hpp>
 #include <stan/math/prim/mat/err/check_square.hpp>
 #include <stan/math/prim/mat/meta/index_type.hpp>
-#include <stan/math/prim/mat/fun/log_determinant_ldlt.hpp>
-#include <stan/math/prim/mat/fun/mdivide_left_ldlt.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/meta/include_summand.hpp>
 #include <stan/math/prim/mat/prob/wishart_rng.hpp>
+#include <stan/math/prim/scal/err/check_greater.hpp>
 
 namespace stan {
   namespace math {
 
     template <class RNG>
-    inline Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>
-    inv_wishart_rng(const double nu,
-                    const Eigen::Matrix
-                    <double, Eigen::Dynamic, Eigen::Dynamic>& S,
-                    RNG& rng) {
+    inline Eigen::MatrixXd
+    inv_wishart_rng(const double nu, const Eigen::MatrixXd& S, RNG& rng) {
       static const char* function("inv_wishart_rng");
 
       using Eigen::MatrixXd;
-
       typename index_type<MatrixXd>::type k = S.rows();
 
-      check_greater(function, "Degrees of freedom parameter", nu, k-1);
+      check_greater(function, "degrees of freedom > dims - 1", nu, k - 1);
       check_square(function, "scale parameter", S);
 
       MatrixXd S_inv = MatrixXd::Identity(k, k);
       S_inv = S.ldlt().solve(S_inv);
 
-      return wishart_rng(nu, S_inv, rng).inverse();
+      MatrixXd asym = wishart_rng(nu, S_inv, rng).inverse();
```

Comment:
When I try that, I get this error:

```
~/cmdstan/stan/lib/stan_math(bugfix/0269-wishart-symm)$ ./runTests.py -j4 test/unit/math/prim/mat/prob/inv_wishart_rng_test.cpp 
------------------------------------------------------------
make -j4 test/unit/math/prim/mat/prob/inv_wishart_rng_test
...
test/unit/math/prim/mat/prob/inv_wishart_rng_test --gtest_output="xml:test/unit/math/prim/mat/prob/inv_wishart_rng_test.xml"
Running main() from gtest_main.cc
[==========] Running 4 tests from 3 test cases.
[----------] Global test environment set-up.
[----------] 1 test from ProbDistributionsInvWishartRng
[ RUN      ] ProbDistributionsInvWishartRng.rng
[       OK ] ProbDistributionsInvWishartRng.rng (2 ms)
[----------] 1 test from ProbDistributionsInvWishartRng (2 ms total)

[----------] 1 test from probdistributionsInvWishartRng
[ RUN      ] probdistributionsInvWishartRng.symmetry
unknown file: Failure
C++ exception with description "invese_spd: matrix not positive definite " thrown in the test body.
[  FAILED  ] probdistributionsInvWishartRng.symmetry (0 ms)
[----------] 1 test from probdistributionsInvWishartRng (0 ms total)

[----------] 2 tests from ProbDistributionsInvWishart
[ RUN      ] ProbDistributionsInvWishart.chiSquareGoodnessFitTest
[       OK ] ProbDistributionsInvWishart.chiSquareGoodnessFitTest (1198 ms)
[ RUN      ] ProbDistributionsInvWishart.SpecialRNGTest
[       OK ] ProbDistributionsInvWishart.SpecialRNGTest (32039 ms)
[----------] 2 tests from ProbDistributionsInvWishart (33237 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 3 test cases ran. (33239 ms total)
[  PASSED  ] 3 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] probdistributionsInvWishartRng.symmetry

 1 FAILED TEST
test/unit/math/prim/mat/prob/inv_wishart_rng_test --gtest_output="xml:test/unit/math/prim/mat/prob/inv_wishart_rng_test.xml" failed
exit now (08/19/16 19:01:39 CEST)
```

The problem is that we can't guarantee that the result is positive definite due to arithmetic issues.  

Any suggestions on what I should do for this case?  I don't want to just fudge the diagonal as that won't give the right random draws.